src/bucket/FutureBucket.cpp

```diff
@@ -233,24 +240,29 @@ FutureBucket::resolve()
     checkState();
     assert(isLive());
     clearInputs();
-    std::shared_ptr<Bucket> bucket;
 
+    if (!mOutputBucket)
     {
         auto timer = LogSlowExecution("Resolving bucket");
-        bucket = mOutputBucket.get();
+        mOutputBucket = mOutputBucketFuture.get();
+
+        // Explicitly reset shared_future to ensure destruction of shared state.
+        // Some compilers store packaged_task lambdas in the shared state,
+        // keeping its captures alive as long as the future is alive.
+        mOutputBucketFuture = std::shared_future<std::shared_ptr<Bucket>>();
     }
 
     if (mOutputBucketHash.empty())
     {
-        mOutputBucketHash = binToHex(bucket->getHash());
+        mOutputBucketHash = binToHex(mOutputBucket->getHash());
```

Comment:
`mOutputBucket` is only set above, right?

So why not setting `mOutputBucketHash` in the same place?
Actually maybe `resolve` should just start with `if (LIVE_OUTPUT) { return mOutputBucket; } ` right now the code is written in an almost idempotent way, which doesn't make much sense